### PR TITLE
Support cross platform vty 6.x with vty-crossplatform

### PR DESCRIPTION
**Pull request by hellwolf**
This is a draft to make the patch visible during fixing [echidna](https://github.com/NixOS/nixpkgs/pull/327511) in nixpkgs.

I have not updated this repo's flake.nix and fix the build yet..

### CODE REVIEW

File: lib/Echidna/UI.hs [hunk 1]
```diff
@@ -243,12 +244,12 @@ stopWorkers :: MonadIO m => [(ThreadId, a)] -> m ()
...
   forM_ workers $ \(threadId, _) -> liftIO $ killThread threadId
 
-vtyConfig :: IO Config
+vtyConfig :: IO VtyUserConfig
 vtyConfig = do
-  config <- Vty.standardIOConfig
-  pure config { inputMap = (Nothing, "\ESC[6;2~", EvKey KPageDown [MShift]) :
-                           (Nothing, "\ESC[5;2~", EvKey KPageUp [MShift]) :
-                           inputMap defaultConfig }
+  config <- Vty.userConfig
```
Comment (hellwolf) [inline lib/Echidna/UI.hs:249]: Note:

The standardIOConfig has some unix specific configurations. From 6.x, it is gone. The closest is a function from vty-unix package.

**Comment by elopez** [comment]
Hi! to avoid duplicating efforts, please take a look at #1245 which also implements the newer vty

**Comment by ggrieco-tob** [comment]
Uhm, I'm wondering if this will solve either #1135  or #1240 :thinking:

**Comment by hellwolf** [comment]
> Hi! to avoid duplicating efforts, please take a look at #1245 which also implements the newer vty

Oh, I see! I will try the patch there and see if it passes the nixpkgs build, then I can abandon my own.

**Comment by hellwolf** [comment]
I got echidna 2.2.3 fixed for nixpkgs.

I saw 2.2.4 release, but that requires a newer commit of hevm, which is not in nixpkgs yet.

**Comment by ggrieco-tob** [comment]
is this still needed?

**Comment by hellwolf** [comment]
As long as  the patch in

https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/tools/security/echidna/default.nix#L11

```
      url = "https://github.com/crytic/echidna/commit/2913b027d7e793390ed489ef6a47d23ec9b3c800.patch";
```

Is still valid?

**Pull request closed without merging**